src/pages/Product/Product.js

```diff
@@ -24,6 +32,54 @@ export class Product extends Component {
       });
   }
 
+  makeCondition = () => {
+    const filterMatch = {
+      bone: 1,
+      hair: 2,
+      growth: 3,
+      skin: 4,
+    };
+
+    const filtered = Object.entries(this.state.filterState).reduce(
+      (acc, [key, value]) => {
+        if (!acc && value) {
+          return acc + `efficacy=${filterMatch[key]}`;
+        }
+
+        if (value) {
+          return acc + `&efficacy=${filterMatch[key]}`;
+        }
+        return acc;
+      },
+      ''
+    );
+
+    fetch(`${GET_PRODUCTS_API}?${filtered}`)
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          productCard: data.message,
+        });
+      });
+    console.log(`filtered>>>>>`, filtered);
+  };
+
+  handleCheckBox = event => {
+    const checkBoxName = event.target.name;
+    const checkBoxNameState = !this.state.filterState[checkBoxName];
+    this.setState(
+      {
+        filterState: {
+          ...this.state.filterState,
+          [checkBoxName]: checkBoxNameState,
+        },
+      },
+      () => {
+        this.makeCondition();
+      }
```

Comment:
이미 makeCondition자체도 함수니까 아래와 같이 표현 가능해보입니다!
```suggestion
  this.makeCondition
```

Reply:
오 !! 저는 화살표 함수로 실행해주어야 동기 문제가 발생하지 않는다고 생각했었어요 !!! 👍🏻